model/post.go

```diff
@@ -166,6 +166,7 @@ type SearchParameter struct {
 	PerPage                *int    `json:"per_page"`
 	IncludeDeletedChannels *bool   `json:"include_deleted_channels"`
 	Modifier               *string `json:"modifier"` // whether it's messages or file
+	HasUserMention         *bool   `json:"has_user_mention"`
```

Comment:
Note that since the files search uses the same type, this will add the parameter to the files search as well. But I did not modify the behaviour of the files search, so this parameter doesn't do anything. What do you think? Should I change the behaviour of the files search if this parameter is passed in? However, I'm not sure what would really be logical behaviour for the files search with this parameter.

Reply:
I wonder if the parameter name is misleading, people might not realise it includes groups because of the inclusion of `user`. What about `include_mentions`?

Reply:
I also find the naming misleading. It's named as if the proposition is "has mention" which implies that the negation of the proposition is "does not have mention" which I would expect to return results with posts that do not contain a mention. 

However, the actual behaviour of the negation is more like "do not apply the proposition *has mention*". 

Put verbosely, the proposition that matches the behaviour is something like "scope to posts with a mention" and the negation is "do not scope to posts with a mention". I would try to find a name that is a closer match to the behaviour.

Reply:
What about `search_user_mentions`? The negation is thus "don't search user mentions" which I think applies here.

Reply:
Yeah, or `scope_to_user_mentions`; 0/5. 

What about to @BenCookie95's question about group mentions? Are those included or *just* user mentions?

Reply:
@cyrilzhang-mm Gotcha. Yeah, maybe we abandon trying to capture the full meaning in the field/JSON key is best and we rather document in in a Godoc on the field and also in the associated Swagger for the API request?